packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Deputy/JsiiTypeAttributeBase.cs

```diff
@@ -17,27 +20,32 @@ protected JsiiTypeAttributeBase(Type nativeType, string fullyQualifiedName)
             Load(nativeType.Assembly);
         }
 
-        void Load(Assembly assembly)
+        private static void Load(Assembly assembly)
         {
-            IEnumerable<Assembly> dependencies = assembly.GetReferencedAssemblies()
-                .Select(assemblyName => Assembly.Load(assemblyName));
+            if (ProcessedAssemblies.Contains(GetAssemblyKey(assembly)))
+            {
+                return;
+            }
 
-            JsiiAssemblyAttribute attribute = assembly.GetCustomAttribute<JsiiAssemblyAttribute>();
+            var attribute = assembly.GetCustomAttribute<JsiiAssemblyAttribute>();
             if (attribute == null)
             {
+                ProcessedAssemblies.Add(GetAssemblyKey(assembly));
                 return;
             }
 
-            foreach (Assembly dependency in dependencies)
+            foreach (var referencedAssembly in assembly.GetReferencedAssemblies())
             {
-                Load(dependency);
+                var loadedReference = Assembly.Load(referencedAssembly);
+                Load(loadedReference);
             }
 
             // find the .tgz resource
             var tarballResourceName = assembly.GetManifestResourceNames().FirstOrDefault(name => name.EndsWith(".tgz"));
             if (tarballResourceName == null)
             {
-                throw new JsiiException("Cannot find embedded tarball resource in assembly " + assembly.GetName(), null);
+                throw new JsiiException("Cannot find embedded tarball resource in assembly " + assembly.GetName(),
```

Comment:
IIRC `JsiiException` is reserved for exceptions that come directly from the JSII runtime. Use a standard exception type or roll your own instead.

Reply:
This was not a change, just a newline added due to a reformat. I will change it in a future PR.

https://github.com/awslabs/jsii/issues/308